### DISCUSSION TITLE
Analysis of Reconstruction Image Instability and RMSE Curve Fluctuations

### DISCUSSION THREAD

**Discussion by Lin-XX-01** [CERN/TIGRE#603] [Q&A]
Dear all,
I have recently been encountering frequent instability in the reconstructed images, where the RMSE curve unexpectedly exhibits sudden jumps, as illustrated in the attached figure. During these jumps, the images develop regions with intensified grayscale. Interestingly, this issue does not occur on other computers. If I may, I would be most grateful for any insights into what might be causing this problem. Thank you very much for your time and assistance.

Thanks
Linxin
![image](https://github.com/user-attachments/assets/b173fc39-9554-473b-9192-46c2080c874e)
![image](https://github.com/user-attachments/assets/d40e76d7-45b2-4f3f-b1d3-c759da4b1dca)
![image](https://github.com/user-attachments/assets/b6424a35-27ff-4092-b6db-f94a4202ee43)

**Comment by AnderBiguri** [comment]
H! Quite strange indeed. 
What algorithm are you using? Can you show the code you used?

**Reply by Lin-XX-01** [reply]
Ok. I suspect there may be a potential issue with the TV denoising function. When I use other functions, such as im3DDenoise, everything seems to work smoothly; however, I encounter some difficulties specifically with the TminimizeAwTV function.
![image](https://github.com/user-attachments/assets/3a51ef07-6950-4269-a22d-f60262f0cac4)
![image](https://github.com/user-attachments/assets/b363c311-9f84-4c33-b169-164ba6cdfa96)

**Reply by AnderBiguri** [reply]
Ah, but this is not an issue then. RMSE between iterations (what you are plotting) basically is a proxy for how big the update size is. For algorithms like SART (or similar) this update size is indeed decreasing with iterations, but this is not true for POCS-like algorithms.

POCS like algorithms (or alternate minimization algorithms) minimize 2 functions alternatively. In this case, likely the TV minimization is pushing the image reconstruction in a relatively opposite direction (not exact opposite, as that is a stopping criteria) to the direction of the data minimization. This leads to a situation where after some TV minimization step, the update size of the data minimization function (SART in this case) now is further from its minima than before, and thus makes a bigger update size than before.

**Reply by AnderBiguri** [reply]
Now, they gray area in the image is indeed weird. Can you tell me what is the specs of the 2 PCs that you have, in where it works for one but not the other?

**Reply by Lin-XX-01** [reply]
Ok. My computer is a 4060 and my partners computer is a 4060ti as well as a 1050ti.

**Reply by AnderBiguri** [reply]
This could be related to #594, hoping to fix soon

**Reply by Lin-XX-01** [reply]
Ok，thank you very much for your time and assistance.

**Reply by zeminLi66** [reply]
Hello, are you also working on industrial CT 3D reconstruction? We use a translational X-ray source to obtain projections, and then perform 3D reconstruction with very limited angles. I wonder if TIGRE code library has this feature.
